src/main/java/core/net/OnlineWorker.java

```diff
@@ -194,6 +194,31 @@ public static List<MatchKurzInfo> getMatchArchive(int teamId, HODateTime firstDa
 		return allMatches;
 	}
 
+	public static boolean downloadMatchData(int matchId, SourceSystem sourceSystem, boolean refresh) {
+		var info = DBManager.instance().loadMatchKurzInfo(sourceSystem, matchId);
+		if (info == null) {
+			var details = OnlineWorker.downloadMatchDetails(sourceSystem, matchId);
+			if (details != null) {
+				info = new MatchKurzInfo();
+				info.setMatchID(details.getMatchID());
+				info.setMatchType(details.getMatchType());
+			}
+			else {
+				return false;
+			}
+		}
+		return downloadMatchData(info, refresh);
+	}
+
+	private static Matchdetails downloadMatchDetails(SourceSystem sourceSystem, int matchId) {
+        var matchDetails = MyConnector.instance().downloadMatchDetails(sourceSystem.to_string(), matchId);
+        if (matchDetails.isEmpty()) {
+            HOLogger.instance().warning(OnlineWorker.class, "Unable to fetch details for match " + matchId);
```

Comment:
I suggest defining a constant for `"Unable to fetch details for match "`.

Reply:
I find this more readable than a constant expression - what would be the advantage of usind a constant instead of the literals?

Reply:
I is not necessarily based using the literal 3 times but the same code 3 times. So the code is it not DRY (Dont Repeat Yourself):
```java
HOLogger.instance().warning(OnlineWorker.class, "Unable to fetch details for match " + matchId);
```

You may use a private static function like this and use it in the 3 places:
```java
	private static void logWarnUnableToFetchDetails(int matchId) {
		HOLogger.instance().warning(OnlineWorker.class, "Unable to fetch details for match " + matchId);
	}
``` 

The benefit is the maintainability of the code in general.